backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java

```diff
@@ -706,44 +711,106 @@ public void attributeUsageWithBracketOperatorExpression() {
     }
-    
+
     /**
      * Testing an OperatorExpression whose operator is an "@", typed by a MetadataDefinition.
      */
     @Test
     public void classificationExpression() {
         MetadataDefinition type = this.builder.createWithName(MetadataDefinition.class, "Syson");
-        
+
+        OperatorExpression classificationExpression = this.builder.create(OperatorExpression.class);
+        classificationExpression.setOperator("@");
+
+        FeatureMembership featureMembership = this.builder.createIn(FeatureMembership.class, classificationExpression);
+        Feature feature = this.builder.create(Feature.class);
+        featureMembership.getOwnedRelatedElement().add(feature);
+
+        this.builder.setType(feature, type);
+
+        this.assertTextualFormEquals("@Syson", classificationExpression);
+    }
+
+    /**
+     * Testing an OperatorExpression whose operator is an "@", typed by a MetadataDefinition. The
+     * classificationExpression can be constructed in different ways that does not impact the way the model is
+     * displayed.
+     */
+    @Test
+    public void classificationExpressionWithArgumentMember() {
+        MetadataDefinition type = this.builder.createWithName(MetadataDefinition.class, "Syson");
+
         OperatorExpression classificationExpression = this.builder.create(OperatorExpression.class);
         classificationExpression.setOperator("@");
-        
+
+        ParameterMembership parameterMembership = this.builder.createIn(ParameterMembership.class, classificationExpression);
+        Feature feature1 = this.builder.create(Feature.class);
+        parameterMembership.getOwnedRelatedElement().add(feature1);
+        FeatureValue featureValue = this.builder.createIn(FeatureValue.class, feature1);
+
+        FeatureReferenceExpression featureReferenceExpression = this.builder.create(FeatureReferenceExpression.class);
+        featureValue.getOwnedRelatedElement().add(featureReferenceExpression);
+        ReturnParameterMembership returnParameterMembership = this.builder.createIn(ReturnParameterMembership.class, featureReferenceExpression);
+        Feature feature2 = this.builder.create(Feature.class);
+        returnParameterMembership.getOwnedRelatedElement().add(feature2);
+
         FeatureMembership featureMembership = this.builder.createIn(FeatureMembership.class, classificationExpression);
         Feature feature = this.builder.create(Feature.class);
         featureMembership.getOwnedRelatedElement().add(feature);
-        
-        builder.setType(feature, type);
+
+        this.builder.setType(feature, type);
 
         this.assertTextualFormEquals("@Syson", classificationExpression);
     }
 
+    /**
+     * Testing an OperatorExpression whose operator is "as".
+     */
+    @Test
+    public void classificationExpressionWithCastOperator() {
+        MetadataDefinition type = this.builder.createWithName(MetadataDefinition.class, "fuel");
```

Comment:
It's seems strange to have a _MetadataDefinition_ here.
Please use "logic" types & variables when you write tests.
In your case you have an _OccurrenceUsage_ (power) on one side and a _MetadataDefinition_ on the other side (fuel).
Please check again and update if you can.

Reply:
Also update every other test in this PR if needed